src/App.tsx

```diff
@@ -1,14 +1,78 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+type Filter = {
+  completed: boolean | null;
+  query: string;
+};
+
+function getFilteredTodos(todos: Todo[], filter: Filter): Todo[] {
+  const filteredTodos = [...todos].filter(todo => {
```

Comment:
```suggestion
  const filteredTodos = todos.filter(todo => {
```
filter method doesn't mutate the original array, so no need to make a copy